File: src/Domain/Models/Challenges/ChallengeSearchItem.cs
using Domain.Entities.Challenges;
using Domain.Enums;

namespace Domain.Models.Challenges;

public class ChallengeSearchItem
{
    public string Id { get; set; }
    public string CreatedByUserId { get; set; }
    public string Name { get; set; }
    public string DescriptionShort { get; set; }
    public double Difficulty { get; set; }
    public ChallengeStatus Status { get; set; }
    public List<TagEntity> Tags { get; set; }
}

File: src/react-frontend/src/app/types/models/challenge/challengeSearchItem.ts
import { ChallengeStatus } from 'app/types/enums/challengeStatus';
import { ChallengeTag } from './challengeTag';

export type ChallengeSearchItem = {
  id: string;
  createdByUserId: string;
  name: string;
  descriptionShort: string;
  difficulty: number;
  status: ChallengeStatus;
  tags: ChallengeTag[];
};
